prow/cmd/deck/pr_history_test.go

```diff
@@ -154,40 +155,62 @@ func TestUpdateCommitData(t *testing.T) {
 	}
 }
 
-func TestParsePullKey(t *testing.T) {
+func TestParsePullURL(t *testing.T) {
 	cases := []struct {
 		name   string
-		key    string
+		addr   string
 		org    string
 		repo   string
 		pr     int
 		expErr bool
 	}{
 		{
-			name: "all good",
-			key:  "kubernetes/test-infra/10169",
+			name: "simple org/repo",
+			addr: "https://prow.k8s.io/pr-history?org=kubernetes&repo=test-infra&pr=10169",
 			org:  "kubernetes",
 			repo: "test-infra",
 			pr:   10169,
 		},
 		{
-			name:   "3rd field needs to be PR number",
-			key:    "kubernetes/test-infra/alpha",
+			name: "Gerrit org/repo",
+			addr: "https://prow.k8s.io/pr-history?org=http://theponyapi.com&repo=test/ponies&pr=12345",
+			org:  "http://theponyapi.com",
+			repo: "test/ponies",
+			pr:   12345,
+		},
+		{
+			name:   "PR needs to be an int",
+			addr:   "https://prow.k8s.io/pr-history?org=kubernetes&repo=test-infra&pr=alpha",
+			expErr: true,
+		},
+		{
+			name:   "missing org",
+			addr:   "https://prow.k8s.io/pr-history?repo=test-infra&pr=10169",
 			expErr: true,
 		},
 		{
-			name:   "not enough parts",
-			key:    "kubernetes/10169",
+			name:   "missing repo",
+			addr:   "https://prow.k8s.io/pr-history?org=kubernetes&pr=10169",
+			expErr: true,
+		},
+		{
+			name:   "missing pr",
+			addr:   "https://prow.k8s.io/pr-history?org=kubernetes&repo=test-infra",
 			expErr: true,
 		},
 	}
 	for _, tc := range cases {
-		org, repo, pr, err := parsePullKey(tc.key)
+		u, err := url.Parse(tc.addr)
+		if err != nil {
+			t.Errorf("bad test URL %s: %v", tc.addr, err)
+			continue
+		}
+		org, repo, pr, err := parsePullURL(u)
 		if (err != nil) != tc.expErr {
-			t.Errorf("%s: unexpected error %v", tc.name, err)
+			t.Errorf("Test %q: unexpected error: %v", tc.name, err)
```

Comment:
Not sure prefixing test case name with `Test ` is useful, not common in our codebase

Reply:
removed